packages/react-router-dom/modules/BrowserRouter.js

```diff
@@ -18,6 +18,12 @@ class BrowserRouter extends React.Component {
   history = createHistory(this.props)
 
   render() {
+    if (this.props.history) {
+      console.error(
+        '`<BrowserRouter history={...}` prop has been ignored. For custom history, ' +
+        'make sure to `import {Router}` and not `import {... as Router}`.')
+    }
+
```

Comment:
This should just use the `warning` package. You'll need to install it for `react-router-dom`, but it is already installed for `react-router`. Basically, just change this block to:
```js
warning(
  !(this.props.history),
  ...
);
```